spanner/session.go

```diff
@@ -924,6 +1004,27 @@ var errGetSessionTimeout = spannerErrorf(codes.Canceled, "timeout / context canc
 // sessions being checked out of the pool.
 func (p *sessionPool) newSessionHandle(s *session) (sh *sessionHandle) {
 	sh = &sessionHandle{session: s, checkoutTime: time.Now(), lastUseTime: time.Now()}
+	if s.isMultiplexed {
+		p.mu.Lock()
+		p.sc.mu.Lock()
+		clientOpt := option.WithGRPCConn(p.sc.connPool.Conn())
+		clientInfo := []string{"gccl", internal.Version}
+		if p.sc.userAgent != "" {
+			agentWithVersion := strings.SplitN(p.sc.userAgent, "/", 2)
+			if len(agentWithVersion) == 2 {
+				clientInfo = append(clientInfo, agentWithVersion[0], agentWithVersion[1])
+			}
+		}
+		p.sc.mu.Unlock()
+		p.mu.Unlock()
+		client, err := vkit.NewClient(context.Background(), clientOpt)
+		if err != nil {
+			return nil
+		}
+		client.SetGoogleClientInfo(clientInfo...)
+		client.CallOptions = s.client.CallOptions
+		sh.client = client
+	}
 	if p.TrackSessionHandles || p.ActionOnInactiveTransaction == Warn || p.ActionOnInactiveTransaction == WarnAndClose || p.ActionOnInactiveTransaction == Close {
```

Comment:
I don't think we want to do this for multiplexed sessions. We keep track of session handles to help debug or clean up leaked sessions. Multiplexed sessions do not leak.

Reply:
I think that overall it might even be better to just create N session handles up front for the multiplexed session when the multiplexed session is created, where N==numChannels. I don't think we need a new session handle for each time one is requested.